internal/terraform/resource.go

```diff
@@ -72,5 +93,5 @@ type resourcesSortedByType []*Resource
 func (a resourcesSortedByType) Len() int      { return len(a) }
 func (a resourcesSortedByType) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
 func (a resourcesSortedByType) Less(i, j int) bool {
-	return a[i].FullType() < a[j].FullType() || (a[i].FullType() == a[j].FullType() && a[i].Mode <= a[j].Mode && a[i].Name <= a[j].Name)
+	return a[i].Mode > a[j].Mode || (a[i].Mode == a[j].Mode && a[i].Spec() < a[j].Spec()) || (a[i].Mode == a[j].Mode && a[i].Spec() == a[j].Spec() && a[i].Name <= a[j].Name)
```

Comment:
Since these conditions became very long and hard to read I'd suggest to break them down into multiple lines:

```suggestion
	if a[i].Mode == a[j].Mode {
		if a[i].Spec() == a[j].Spec() {
			return a[i].Name <= a[j].Name
		}
		return a[i].Spec() < a[j].Spec()
	}
	return a[i].Mode > a[j].Mode
```